web-platform/website/src/components/pages/account-settings/EditAccountPasswordView.tsx

```diff
@@ -0,0 +1,391 @@
+  validatePasswordRules
+} from "../login/LoginFormValidator";
+import {
+  AccountSettingsState,
+  useAccountSettingsDataContext,
+} from "./AccountSettingsDataContext";
+import { PasswordRules } from "../login/PasswordRules";
+
+const strings = Strings.components.pages.accountSettings.editPasswordAccount;
+const firebaseStings = Strings.utils.firebaseError.errorMessages;
+
+export function EditAccountPasswordView(props:any) {
+  const {showToast} = props
+  const [password, setPassword] = useState({
+    current: "",
+    new: "",
+    confirm: ""
+  });
+
+  const [passwordVisible, setPasswordVisible] = useState({
+    current: false,
+    new: false,
+    confirm: false
+  });
+
+  const [eyeIconVisible, setEyeIconVisible] = useState({
+    current: false,
+    new: false,
+    confirm: false
+  });
+  const [validationResult, setValidationResult] =
+    useState<PasswordAccountValidationResult>();
+  const { setAccountSettingsState } = useAccountSettingsDataContext();
+  const { changeUserPassword } = useAuthentication();
+
+  const isCurrentInvalid = Boolean(
+    validationResult &&
+    validationResult.currentPassword &&
+    validationResult.currentPassword.error
+  );
+
+  const isNewInvalid = Boolean(
+    validationResult &&
+    validationResult.newPassword &&
+    validationResult.newPassword.error
+  );
+
+  const isConfirmInvalid = Boolean(
+    validationResult &&
+    validationResult.confirmPassword &&
+    validationResult.confirmPassword.error
+  );
+
+  const isDisabled = !password.current || !password.new || !password.confirm;
+
+  const currentRef = useRef(password.current);
+  currentRef.current = password.current;
+
+  const newRef = useRef(password.new);
+  newRef.current = password.new;
+
+  const confirmRef = useRef(password.confirm);
+  confirmRef.current = password.confirm;
+
+  useEffect(() => {
+    const rules = validatePasswordRules(password.new);
+    const isValid = !rules.find((rule) => !rule.valid);
+    setValidationResult((validationResult) => ({
+      ...(validationResult || {
+        currentPassword: { error: undefined },
+        newPassword: { error: undefined },
+        confirmPassword: { error: undefined },
+      }),
+      valid: isValid,
+      passwordRules: (newRef.current && rules) || [],
+    }));
+  }, [password.new]);
+
+  async function validateAndUpdate() {
+
+    const hideEyeicon = {
+      current: false,
+      new: false,
+      confirm: false
+    }
+    const validationResult = validatePasswordAccountForm(
+      currentRef.current,
+      newRef.current,
+      confirmRef.current,
+    );
+
+    setValidationResult(validationResult);
+
+    if (!validationResult || !validationResult.valid) {
+      setEyeIconVisible(hideEyeicon);
+      //when user enters new passsword breaking password rules
+      if (!validationResult.newPassword.error) {
+        const passwordErr = "Invalid password. Requires 1 lowercase, 1 uppercase, 1 number, at least 8 characters";
```

Comment:
Strings file

Reply:
Also we have a control that checks off rules for password. We should likely use that here?

Reply:
Yes, had to update to highlight form errors